avalon_listener/Dockerfile

```diff
@@ -0,0 +1,196 @@
+# Note: The latter will compile in HW or SIM mode depending on the
+# availability of /dev/isgx and /var/run/aesmd/aesm.socket
+RUN OPENSSL_VER=1.1.1d \
+ && wget https://www.openssl.org/source/openssl-$OPENSSL_VER.tar.gz \
+ && tar -zxf openssl-$OPENSSL_VER.tar.gz \
+ && cd openssl-$OPENSSL_VER/ \
+ && ./config \
+ && THREADS=8 \
+ && make -j$THREADS \
+ && make test \
+ && make install -j$THREADS \
+ && ldconfig \
+ && ln -s /etc/ssl/certs/* /usr/local/ssl/certs/ \
+ && cd .. \
+ && rm -rf openssl-$OPENSSL_VER \
+ && git clone https://github.com/intel/intel-sgx-ssl.git  \
+ && . /opt/intel/sgxsdk/environment \
+ && (cd intel-sgx-ssl/openssl_source; mv /tmp/openssl-$OPENSSL_VER.tar.gz . ) \
+ && (cd intel-sgx-ssl/Linux; \
+    if ([ -c /dev/isgx ] && [ -S /var/run/aesmd/aesm.socket ]); then SGX_MODE=HW; \
+    else SGX_MODE=SIM; \
+    fi; \
+    make SGX_MODE=${SGX_MODE} DESTDIR=/opt/intel/sgxssl all test ) \
+ && (cd intel-sgx-ssl/Linux; make install ) \
+ && rm -rf /tmp/intel-sgx-ssl \
+ && echo "SGX_SSL=/opt/intel/sgxssl" >> /etc/environment
+
+COPY . /project/avalon
+
+ARG SGX_MODE
+ARG MAKECLEAN
+ARG TCF_DEBUG_BUILD
+ARG DISPLAY
+ARG XAUTHORITY
+
+# Environment setup
+
+RUN echo "SOLC_BINARY=/root/.py-solc/solc-v0.4.25/bin/solc" >> /etc/environment \
+ && echo "WALLET_PRIVATE_KEY=B413189C95B48737AE2D9AF4CAE97EB03F4DE40599DF8E6C89DCE4C2E2CBA8DE" >> /etc/environment \
+ && echo "TCF_HOME=/project/avalon/" >> /etc/environment \
+ && if [ ! -z "$http_proxy"  ]; then \
+    echo 'Acquire::http::Proxy "'$http_proxy'";' >> /etc/apt/apt.conf.d/00proxy; \
+        echo "http_proxy=$http_proxy" >> /etc/wgetrc; \
+        echo "http_proxy=$http_proxy" >> /etc/environment; \
+        echo "HTTP_PROXY=$(echo $http_proxy | sed 's,[a-zA-Z]*://,,')" >> /etc/environment; \
+    fi \
+ && if [ ! -z "$ftp_proxy"  ];  then \
+    echo 'Acquire::ftp::Proxy "'$ftp_proxy'";' >> /etc/apt/apt.conf.d/00proxy; \
+        echo "ftp_proxy=$ftp_proxy" >> /etc/wgetrc; \
+        echo "ftp_proxy=$ftp_proxy" >> /etc/environment; \
+    fi \
+ && if [ ! -z "$https_proxy" ]; then \
+    echo 'Acquire::https::Proxy "'$https_proxy'";' >> /etc/apt/apt.conf.d/00proxy; \
+        echo "https_proxy=$https_proxy" >> /etc/wgetrc; \
+        echo "https_proxy=$https_proxy" >> /etc/environment; \
+        echo "HTTPS_PROXY=$(echo $https_proxy | sed 's,[a-zA-Z]*://,,')" >> /etc/environment; \
+    fi
+
+ENV http_proxy=$http_proxy
+ENV https_proxy=$https_proxy
+ENV TCF_HOME=/project/avalon
+ENV SGX_SSL=/opt/intel/sgxssl
+ENV SGX_SDK=/opt/intel/sgxsdk
+ENV PATH=$PATH:$SGX_SDK/bin:$SGX_SDK/bin/x64
+ENV PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$SGX_SDK/pkgconfig
+ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$SGX_SDK/sdk_libs
+# Solc compiler path to compile solidity contract code
+ENV SOLC_BINARY=/root/.py-solc/solc-v0.4.25/bin/solc
+ENV TCF_ENCLAVE_CODE_SIGN_PEM="$TCF_HOME/enclave.pem"
+
+RUN openssl genrsa -3 -out $TCF_HOME/enclave.pem 3072
+
+WORKDIR /project/avalon/tc/sgx/common

```

Comment:
Listener shouldn't depend on tc/sgx and not build tc/sgx

Reply:
Listener is dependent on common/python (python SDK) and common/python is dependent on tc/sgx. In coming sprints, common/python, tc/sgx and other components probably will have its own Dockerfile to build the package which will then be used in listener.

Reply:
My understanding is listener is completely implemented in python and should not depend on anything on tc/sgx.  Which part of common/python related to listener is dependent on tc/sgx ? currently common/python module is built separately and not required to build from avalon_listerner Dockerfile. Were you getting any errors if tc/sgx is not included in listener. @manju956 @Ram-srini  Could you please provide your inputs ?

Reply:
listener is not dependent on tc/sgx but common/python is. Currently, common/python is not build seperately from listener. this will be done in coming sprints. Probably common/python will have its seperate Dockerfile.

Reply:
ok. You mean listener dockerimage needs common/python modules and that's why common/python (and tc/sgx) is build in listener docker image ? common/python has a separate Makefile - https://github.com/hyperledger/avalon/blob/master/examples/common/python/Makefile. So if its not a big deal let's create a docker image for common python which can be used by avalon listener. That will avoid doing rework on avalon listerner module later and it'll also help cleanly separate out avalon listener

Reply:
If it's too much of a work to do in this PR, please add a github issue or Jira ticket to track this as a separate PR

Reply:
This is part of the activity to break down avalon into different components, creating python/C++ SDKs etc.

Reply:
Cool. Please capture it somewhere in the activity breakdown sheet. What I mean is we still need to rework on avalon listener even if we merge this PR in the current state. So all pending/dependent tasks for Avalon listener could be captured somewhere for ease of tracking. Else after this PR is merged/closed, the open comments are not easy to track.

Reply:
@manojgop As we discussed it many a times the overall goal of the activity, this is just one part of it. Currently Avalon is designed in such a way that the components are tightly coupled to each other. It becomes complex to seperate out multiple components in one go and you are aware about it. We take one component at a time, work on it. Then we discuss about next and start work on that and so on.

Reply:
I understand that. But review comments captured in PR which will not be addressed in this PR has to be captured somewhere for tracking. Another option is we don't merge this PR till we raise a new PR to separate dependent components like common/python as a separate component.

Reply:
Not merging this PR will make things even more difficult as listener and enclave manager have dependencies on common/python. We are going to address seperating out other parts on avalon in coming sprints.

Reply:
Looks like it make more sense to separate common/python as an independent component first and then to separate listener and enclave manager to reduce rework. We can keep this PR open and first work on common/python if that approach is better.